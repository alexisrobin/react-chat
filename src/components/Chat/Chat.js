import React from 'react';
import { Container, Row } from 'reactstrap';
import './Chat.scss';
import Conversation from '../Conversation';
import MessageInput from '../MessageInput';

const Chat = ({ messages, users, currentUserId, addMessage }) => {
    return (
        <Container>
            <Row className="chat-header p-1">
                {users.filter(user => user.id === currentUserId).shift().name}
            </Row>
            <Row>
                <Conversation
                    messages={messages}
                    users={users}
                    currentUserId={currentUserId} />
            </Row>
            <Row>
                <MessageInput
                    addMessage={addMessage}
                    currentUserId={currentUserId} />
            </Row>
        </Container>
    )
}

export default Chat;