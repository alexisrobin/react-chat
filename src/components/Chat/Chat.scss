@import '../../const.scss';

.chat-header {
    border-bottom: 1px #ddd solid;
    justify-content: center;
    color: $secondary;
    font-size: 2.5rem;
}