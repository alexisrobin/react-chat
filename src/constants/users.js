export const users = [
    {
        id: 1,
        name: "Chat A"
    },
    {
        id: 2,
        name: "Chat B"
    }
]