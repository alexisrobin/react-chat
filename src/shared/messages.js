export const messages = [
    {
        "id": 0,
        "senderId": 1,
        "content": "Mollit minim adipisicing tempor reprehenderit. Eu aliquip magna aliquip eiusmod. Aliquip anim exercitation elit eiusmod aute magna amet anim proident. Tempor ut enim ullamco elit veniam cillum cupidatat sint. Aliquip pariatur culpa amet non aliquip anim non dolore elit. Incididunt amet Lorem occaecat cupidatat officia dolor deserunt tempor nisi aute occaecat non ut. Cillum duis exercitation adipisicing reprehenderit quis ipsum veniam id commodo enim qui amet duis ut.\r\n",
        "timestamp": "2016-10-30T10:28:05"
    },
    {
        "id": 1,
        "senderId": 2,
        "content": "Id velit aute elit ipsum anim officia. Non incididunt minim ipsum deserunt nulla qui labore ex. Dolore aliquip et amet aute occaecat deserunt deserunt Lorem mollit do irure. Consectetur eu adipisicing excepteur in aliquip quis do ut.\r\n",
        "timestamp": "2015-02-24T03:42:08"
    },
    {
        "id": 2,
        "senderId": 1,
        "content": "Exercitation ad nostrud et proident reprehenderit enim esse in qui voluptate ea. Aliquip mollit esse sunt quis consequat exercitation consectetur sunt laboris nisi ipsum. Mollit occaecat deserunt occaecat veniam exercitation labore adipisicing aliquip dolore culpa officia nostrud ea. Lorem commodo ad nisi consectetur ad nostrud.\r\n",
        "timestamp": "2016-04-29T05:57:31"
    },
    {
        "id": 3,
        "senderId": 1,
        "content": "In ut sint ut ex laborum magna dolore do est. Deserunt dolore minim consectetur nostrud dolor. Fugiat tempor sit aliquip adipisicing ea ex mollit mollit commodo Lorem culpa in ad. Deserunt consectetur deserunt sit enim ea. Cupidatat exercitation irure irure eu. Deserunt incididunt minim magna in qui proident reprehenderit Lorem enim proident id excepteur excepteur. Occaecat ullamco est dolor commodo eu est dolore.\r\n",
        "timestamp": "2016-02-06T07:55:13"
    },
    {
        "id": 4,
        "senderId": 1,
        "content": "Do deserunt labore ullamco aute pariatur quis elit ad excepteur ut ex magna. Ullamco in eiusmod consequat proident occaecat do nisi commodo. Esse ex consequat aliquip pariatur voluptate est magna. Cillum consequat excepteur aliqua occaecat ea id anim eiusmod laborum sit.\r\n",
        "timestamp": "2015-01-05T05:24:06"
    },
    {
        "id": 5,
        "senderId": 2,
        "content": "Exercitation aliqua sit ullamco mollit fugiat elit do velit ex voluptate officia cupidatat aliqua. Laborum enim eiusmod occaecat laboris eiusmod exercitation non cillum sint do. Ipsum cillum est duis ad non tempor consequat Lorem.\r\n",
        "timestamp": "2016-06-25T11:16:49"
    },
    {
        "id": 6,
        "senderId": 1,
        "content": "Quis ea ut minim qui officia excepteur cillum commodo adipisicing veniam duis. Do eu ex exercitation culpa sint magna voluptate. Lorem mollit nostrud minim magna ea non duis incididunt nulla. Nostrud ea velit duis exercitation. Lorem ad reprehenderit ea ea non excepteur sint magna culpa laborum ex ad duis. Est est ut eiusmod officia duis aliqua proident et aliqua do adipisicing culpa.\r\n",
        "timestamp": "2016-01-01T10:05:01"
    },
    {
        "id": 7,
        "senderId": 2,
        "content": "Nulla pariatur nulla cupidatat duis elit dolore tempor velit excepteur. Officia ipsum irure exercitation aliqua enim elit. Cupidatat deserunt ad culpa excepteur voluptate.\r\n",
        "timestamp": "2016-07-04T11:24:32"
    },
    {
        "id": 8,
        "senderId": 2,
        "content": "Non exercitation velit veniam aliqua ut anim. Voluptate id quis ea officia eiusmod et do nisi ut fugiat tempor. Fugiat occaecat reprehenderit et ea aute eu proident laboris. Aliqua consequat adipisicing deserunt labore aliquip fugiat ad irure eu fugiat do deserunt. Quis exercitation nulla sit id labore.\r\n",
        "timestamp": "2018-03-30T08:28:05"
    },
    {
        "id": 9,
        "senderId": 1,
        "content": "Tempor in culpa veniam nostrud. Laboris aliqua ea ea anim labore proident et duis id. Nulla do mollit minim sunt. Sunt officia aliqua minim anim. Anim commodo id dolor elit est quis Lorem nostrud fugiat reprehenderit anim. Proident nulla laborum do sit velit sint exercitation ullamco deserunt laborum et in voluptate consectetur.\r\n",
        "timestamp": "2014-02-09T08:32:25"
    },
    {
        "id": 10,
        "senderId": 1,
        "content": "Dolore est cupidatat id ullamco. Eiusmod eu commodo mollit irure qui velit eiusmod irure laborum duis irure ullamco cillum aliqua. Ex irure fugiat aliqua cupidatat id reprehenderit ipsum nostrud qui cillum elit.\r\n",
        "timestamp": "2017-04-16T09:37:54"
    },
    {
        "id": 11,
        "senderId": 1,
        "content": "Quis exercitation deserunt anim amet. Deserunt excepteur id laboris duis deserunt dolore esse ad duis cupidatat labore ex eiusmod. Ullamco excepteur Lorem quis adipisicing aliqua nisi anim officia consequat irure incididunt. Do mollit adipisicing ipsum et. Amet aliqua incididunt ex officia mollit est laboris occaecat excepteur enim ea aliquip tempor. Ea ut sint adipisicing deserunt cillum fugiat commodo proident tempor aliquip adipisicing incididunt sint.\r\n",
        "timestamp": "2017-12-11T11:11:40"
    },
    {
        "id": 12,
        "senderId": 2,
        "content": "Nisi ad excepteur minim excepteur nostrud in non nisi labore ea deserunt culpa sit deserunt. Do irure irure aute quis veniam ad voluptate enim. Excepteur do irure minim magna ea anim nisi et laborum id irure deserunt esse. Irure veniam eiusmod consequat voluptate ad quis do consectetur cillum irure exercitation consequat. Dolore esse occaecat amet in.\r\n",
        "timestamp": "2015-01-22T09:12:28"
    },
    {
        "id": 13,
        "senderId": 2,
        "content": "Ullamco culpa dolor est reprehenderit ullamco exercitation. Incididunt velit id irure proident mollit ut adipisicing Lorem anim adipisicing non cupidatat. Occaecat laboris consequat consequat esse ad consequat laboris ex sunt nostrud adipisicing anim officia. Aliquip amet id duis est aute. Sint cupidatat amet non ut excepteur aliqua magna aute ipsum deserunt amet est nostrud.\r\n",
        "timestamp": "2015-08-20T12:17:56"
    },
    {
        "id": 14,
        "senderId": 1,
        "content": "Qui et id tempor tempor adipisicing ad velit aute. Sit eiusmod non voluptate ut amet sunt. Reprehenderit magna ea voluptate tempor dolore veniam ut enim deserunt nulla. Tempor minim nostrud do sit Lorem non fugiat eu sint ullamco occaecat reprehenderit. Ad id sint pariatur commodo do. Dolore aliqua nostrud incididunt amet fugiat.\r\n",
        "timestamp": "2018-01-27T12:15:51"
    },
    {
        "id": 15,
        "senderId": 2,
        "content": "Nisi Lorem nostrud nulla incididunt laborum velit incididunt fugiat ullamco eu minim et duis culpa. Quis aute ad dolor do aliquip tempor aute nostrud. Sit labore nisi ipsum magna eu voluptate sint eiusmod cupidatat proident et mollit consectetur.\r\n",
        "timestamp": "2015-05-02T11:58:37"
    }
]