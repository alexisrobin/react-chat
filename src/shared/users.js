export const users = [
    {
        id: 1,
        name: "Dagobert"
    },
    {
        id: 2,
        name: "Childebert"
    }
]