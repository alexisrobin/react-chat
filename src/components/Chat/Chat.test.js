import React from 'react';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import Chat from './Chat';
import { messages } from '../../constants/messages';
import { users } from '../../constants/users';

describe('<Chat />', () => {
    it('should match its reference snapshot', () => {
        const wrapper = shallow(
            <Chat 
                messages={messages}
                users={users}
                currentUserId={1}/>
        );

        expect(wrapper).to.matchSnapshot()
    })
})