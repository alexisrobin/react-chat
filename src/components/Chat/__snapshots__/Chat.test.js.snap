// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> should match its reference snapshot 1`] = `
<Container
  tag="div"
>
  <Row
    className="chat-header p-1"
    tag="div"
  >
    Chat A
  </Row>
  <Row
    tag="div"
  >
    <Conversation
      currentUserId={1}
      messages={
        Array [
          Object {
            "content": "Mollit minim adipisicing tempor reprehenderit. Eu aliquip magna aliquip eiusmod. Aliquip anim exercitation elit eiusmod aute magna amet anim proident. Tempor ut enim ullamco elit veniam cillum cupidatat sint. Aliquip pariatur culpa amet non aliquip anim non dolore elit. Incididunt amet Lorem occaecat cupidatat officia dolor deserunt tempor nisi aute occaecat non ut. Cillum duis exercitation adipisicing reprehenderit quis ipsum veniam id commodo enim qui amet duis ut.
",
            "id": 0,
            "senderId": 1,
            "timestamp": "2016-10-30T10:28:05",
          },
          Object {
            "content": "Id velit aute elit ipsum anim officia. Non incididunt minim ipsum deserunt nulla qui labore ex. Dolore aliquip et amet aute occaecat deserunt deserunt Lorem mollit do irure. Consectetur eu adipisicing excepteur in aliquip quis do ut.
",
            "id": 1,
            "senderId": 2,
            "timestamp": "2015-02-24T03:42:08",
          },
          Object {
            "content": "Exercitation ad nostrud et proident reprehenderit enim esse in qui voluptate ea. Aliquip mollit esse sunt quis consequat exercitation consectetur sunt laboris nisi ipsum. Mollit occaecat deserunt occaecat veniam exercitation labore adipisicing aliquip dolore culpa officia nostrud ea. Lorem commodo ad nisi consectetur ad nostrud.
",
            "id": 2,
            "senderId": 1,
            "timestamp": "2016-04-29T05:57:31",
          },
          Object {
            "content": "In ut sint ut ex laborum magna dolore do est. Deserunt dolore minim consectetur nostrud dolor. Fugiat tempor sit aliquip adipisicing ea ex mollit mollit commodo Lorem culpa in ad. Deserunt consectetur deserunt sit enim ea. Cupidatat exercitation irure irure eu. Deserunt incididunt minim magna in qui proident reprehenderit Lorem enim proident id excepteur excepteur. Occaecat ullamco est dolor commodo eu est dolore.
",
            "id": 3,
            "senderId": 1,
            "timestamp": "2016-02-06T07:55:13",
          },
          Object {
            "content": "Do deserunt labore ullamco aute pariatur quis elit ad excepteur ut ex magna. Ullamco in eiusmod consequat proident occaecat do nisi commodo. Esse ex consequat aliquip pariatur voluptate est magna. Cillum consequat excepteur aliqua occaecat ea id anim eiusmod laborum sit.
",
            "id": 4,
            "senderId": 1,
            "timestamp": "2015-01-05T05:24:06",
          },
          Object {
            "content": "Exercitation aliqua sit ullamco mollit fugiat elit do velit ex voluptate officia cupidatat aliqua. Laborum enim eiusmod occaecat laboris eiusmod exercitation non cillum sint do. Ipsum cillum est duis ad non tempor consequat Lorem.
",
            "id": 5,
            "senderId": 2,
            "timestamp": "2016-06-25T11:16:49",
          },
          Object {
            "content": "Quis ea ut minim qui officia excepteur cillum commodo adipisicing veniam duis. Do eu ex exercitation culpa sint magna voluptate. Lorem mollit nostrud minim magna ea non duis incididunt nulla. Nostrud ea velit duis exercitation. Lorem ad reprehenderit ea ea non excepteur sint magna culpa laborum ex ad duis. Est est ut eiusmod officia duis aliqua proident et aliqua do adipisicing culpa.
",
            "id": 6,
            "senderId": 1,
            "timestamp": "2016-01-01T10:05:01",
          },
          Object {
            "content": "Nulla pariatur nulla cupidatat duis elit dolore tempor velit excepteur. Officia ipsum irure exercitation aliqua enim elit. Cupidatat deserunt ad culpa excepteur voluptate.
",
            "id": 7,
            "senderId": 2,
            "timestamp": "2016-07-04T11:24:32",
          },
          Object {
            "content": "Non exercitation velit veniam aliqua ut anim. Voluptate id quis ea officia eiusmod et do nisi ut fugiat tempor. Fugiat occaecat reprehenderit et ea aute eu proident laboris. Aliqua consequat adipisicing deserunt labore aliquip fugiat ad irure eu fugiat do deserunt. Quis exercitation nulla sit id labore.
",
            "id": 8,
            "senderId": 2,
            "timestamp": "2018-03-30T08:28:05",
          },
          Object {
            "content": "Tempor in culpa veniam nostrud. Laboris aliqua ea ea anim labore proident et duis id. Nulla do mollit minim sunt. Sunt officia aliqua minim anim. Anim commodo id dolor elit est quis Lorem nostrud fugiat reprehenderit anim. Proident nulla laborum do sit velit sint exercitation ullamco deserunt laborum et in voluptate consectetur.
",
            "id": 9,
            "senderId": 1,
            "timestamp": "2014-02-09T08:32:25",
          },
          Object {
            "content": "Dolore est cupidatat id ullamco. Eiusmod eu commodo mollit irure qui velit eiusmod irure laborum duis irure ullamco cillum aliqua. Ex irure fugiat aliqua cupidatat id reprehenderit ipsum nostrud qui cillum elit.
",
            "id": 10,
            "senderId": 1,
            "timestamp": "2017-04-16T09:37:54",
          },
          Object {
            "content": "Quis exercitation deserunt anim amet. Deserunt excepteur id laboris duis deserunt dolore esse ad duis cupidatat labore ex eiusmod. Ullamco excepteur Lorem quis adipisicing aliqua nisi anim officia consequat irure incididunt. Do mollit adipisicing ipsum et. Amet aliqua incididunt ex officia mollit est laboris occaecat excepteur enim ea aliquip tempor. Ea ut sint adipisicing deserunt cillum fugiat commodo proident tempor aliquip adipisicing incididunt sint.
",
            "id": 11,
            "senderId": 1,
            "timestamp": "2017-12-11T11:11:40",
          },
          Object {
            "content": "Nisi ad excepteur minim excepteur nostrud in non nisi labore ea deserunt culpa sit deserunt. Do irure irure aute quis veniam ad voluptate enim. Excepteur do irure minim magna ea anim nisi et laborum id irure deserunt esse. Irure veniam eiusmod consequat voluptate ad quis do consectetur cillum irure exercitation consequat. Dolore esse occaecat amet in.
",
            "id": 12,
            "senderId": 2,
            "timestamp": "2015-01-22T09:12:28",
          },
          Object {
            "content": "Ullamco culpa dolor est reprehenderit ullamco exercitation. Incididunt velit id irure proident mollit ut adipisicing Lorem anim adipisicing non cupidatat. Occaecat laboris consequat consequat esse ad consequat laboris ex sunt nostrud adipisicing anim officia. Aliquip amet id duis est aute. Sint cupidatat amet non ut excepteur aliqua magna aute ipsum deserunt amet est nostrud.
",
            "id": 13,
            "senderId": 2,
            "timestamp": "2015-08-20T12:17:56",
          },
          Object {
            "content": "Qui et id tempor tempor adipisicing ad velit aute. Sit eiusmod non voluptate ut amet sunt. Reprehenderit magna ea voluptate tempor dolore veniam ut enim deserunt nulla. Tempor minim nostrud do sit Lorem non fugiat eu sint ullamco occaecat reprehenderit. Ad id sint pariatur commodo do. Dolore aliqua nostrud incididunt amet fugiat.
",
            "id": 14,
            "senderId": 1,
            "timestamp": "2018-01-27T12:15:51",
          },
          Object {
            "content": "Nisi Lorem nostrud nulla incididunt laborum velit incididunt fugiat ullamco eu minim et duis culpa. Quis aute ad dolor do aliquip tempor aute nostrud. Sit labore nisi ipsum magna eu voluptate sint eiusmod cupidatat proident et mollit consectetur.
",
            "id": 15,
            "senderId": 2,
            "timestamp": "2015-05-02T11:58:37",
          },
        ]
      }
      users={
        Array [
          Object {
            "id": 1,
            "name": "Chat A",
          },
          Object {
            "id": 2,
            "name": "Chat B",
          },
        ]
      }
    />
  </Row>
  <Row
    tag="div"
  >
    <MessageInput
      currentUserId={1}
    />
  </Row>
</Container>
`;
